File: app/src/main/java/com/example/argohunt/models/ArgoHunt.java
package com.example.argohunt.models;

import java.util.Date;
import java.util.UUID;

public class ArgoHunt
{
    private UUID mId;
    private String mHuntTitle;
    private Date mDate;
    private boolean mLocationFound;

    public ArgoHunt()
    {
        this.mId = UUID.randomUUID();
        this.mDate = new Date();

    }

    public UUID getmId()
    {
        return mId;
    }

    public String getmHuntTitle()
    {
        return mHuntTitle;
    }

    public void setmHuntTitle(String mHuntTitle)
    {
        this.mHuntTitle = mHuntTitle;
    }

    public Date getmDate()
    {
        return mDate;
    }

    public void setmDate(Date mDate)
    {
        this.mDate = mDate;
    }

    public boolean ismLocationFound()
    {
        return mLocationFound;
    }

    public void setmLocationFound(boolean mLocationFound)
    {
        this.mLocationFound = mLocationFound;
    }
}

File: app/src/main/java/com/example/argohunt/models/ArgoHunt.kt
package com.example.argohunt.models

import java.util.*

class ArgoHunt
{
    private val mId: UUID
    private var mHuntTitle: String? = null
    private var mDate: Date
    private var mLocationFound = false

    fun getmId(): UUID
    {
        return mId
    }

    fun getmHuntTitle(): String?
    {
        return mHuntTitle
    }

    fun setmHuntTitle(mHuntTitle: String?)
    {
        this.mHuntTitle = mHuntTitle
    }

    fun getmDate(): Date
    {
        return mDate
    }

    fun setmDate(mDate: Date)
    {
        this.mDate = mDate
    }

    fun ismLocationFound(): Boolean
    {
        return mLocationFound
    }

    fun setmLocationFound(mLocationFound: Boolean)
    {
        this.mLocationFound = mLocationFound
    }

    init
    {
        mId = UUID.randomUUID()
        mDate = Date()
    }
}